tests/e2e/vmgroups_test.go

```diff
@@ -431,3 +434,175 @@ func (vg *VmGroupTest) TestVmGroupVolumeClone(c *C) {
+	out, err = dockercli.ExecContainer(vg.config.DockerHosts[0], vg.volName1, vg.testContainer)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Docker may not have completed the detach yet with the host.
+	misc.SleepForSec(2)
+
+	// Status should be detached
+	status := verification.VerifyDetachedStatus(vg.volName1, vg.config.DockerHosts[0], vg.config.EsxHost)
+	c.Assert(status, Equals, true, Commentf("Volume %s is not detached", vg.volName1))
+
+	// Verify volume can be mounted and unmounted for the second volume
+	out, err = dockercli.ExecContainer(vg.config.DockerHosts[0], vg.volName2+"@"+vg.config.Datastores[1], vg.testContainer)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Docker may not have completed the detach yet with the host.
+	misc.SleepForSec(2)
+
+	// Status should be detached
+	status = verification.VerifyDetachedStatusOnNonDefaultDS(vg.volName2, vg.config.Datastores[1],
+		vg.config.DockerHosts[0], vg.config.EsxHost)
+	c.Assert(status, Equals, true, Commentf("Volume %s is not detached", vg.volName2+"@"+vg.config.Datastores[1]))
+
+	// remove VM1 and VM2 from vgTestVMgroup1 and then remove the vmgroup
+	out, err = adminutils.RemoveVMFromVMgroup(vg.config.EsxHost, vgTestVMgroup1, vg.config.DockerHostNames[0])
+	c.Assert(err, IsNil, Commentf(out))
+
+	out, err = adminutils.RemoveVMFromVMgroup(vg.config.EsxHost, vgTestVMgroup1, vg.config.DockerHostNames[1])
+	c.Assert(err, IsNil, Commentf(out))
+
+	log.Printf("Removing test vmgroup %s", vgTestVMgroup1)
+	out, err = adminutils.DeleteVMgroup(vg.config.EsxHost, vgTestVMgroup1, true)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// vmgroup has been removed and all volumes belong to it have been removed
+	// "err.Error()" will be filled with "exit status 1"
+	out, err = verification.GetVMGroupForVolume(vg.config.EsxHost, vg.volName1)
+	log.Println("GetVMGroupForVolume return out[%s] err[%s] for volume %s", out, err, vg.volName1)
+	c.Assert(err.Error(), Equals, "exit status 1", Commentf("volume %s should be removed", vg.volName1))
+
+	out, err = verification.GetVMGroupForVolume(vg.config.EsxHost, vg.volName2)
+	log.Println("GetVMGroupForVolume return out[%s] err[%s] for volume %s", out, err, vg.volName2)
+	c.Assert(err.Error(), Equals, "exit status 1", Commentf("volume %s should be removed", vg.volName2))
+
+	// Recreate the test VM group1 and add VM1, VM2 back to the vmgroup
+	cmd = adminconst.CreateVMgroup + vgTestVMgroup1 + " --default-datastore " + vg.config.Datastores[0]
+	log.Printf("Creating test vmgroup %s", vgTestVMgroup1)
+	out, err = ssh.InvokeCommand(vg.config.EsxHost, cmd)
+	c.Assert(err, IsNil, Commentf(out))
+
+	out, err = adminutils.AddVMToVMgroup(vg.config.EsxHost, vgTestVMgroup1, vg.config.DockerHostNames[0])
+	c.Assert(err, IsNil, Commentf(out))
+
+	out, err = adminutils.AddVMToVMgroup(vg.config.EsxHost, vgTestVMgroup1, vg.config.DockerHostNames[1])
+	c.Assert(err, IsNil, Commentf(out))
+	misc.LogTestEnd(c.TestName())
```

Comment:
Why do we need to recreate and add the VMs back? is it for maintaining the test config?